test-network-function/networking/suite.go

```diff
@@ -275,3 +300,123 @@ func testNodePort(env *config.TestEnvironment) {
+	var result []PortDeclared
+	protocolName, port, name := "", "", ""
+	ocCommandToExecute := fmt.Sprintf(CommandPortDeclared, podName, podNamespace, container)
+	res, _ := utils.ExecuteCommand(ocCommandToExecute, ocCommandTimeOut, interactive.GetContext(false))
+	x := strings.Split(res, "\n")
+	for _, i := range x {
+		if strings.Contains(i, "containerPort") {
+			s := strings.Split(i, ":")
+			s = strings.Split(s[1], ",")
+			port = s[0]
+		}
+		if strings.Contains(i, "name") {
+			s := strings.Split(i, ":")
+			s = strings.Split(s[1], ",")
+			name = s[0]
+		}
+		if strings.Contains(i, "protocol") {
+			s := strings.Split(i, ":")
+			protocolName = s[1]
+		}
+		if port == "" || name == "" || protocolName == "" {
+			continue
+		}
+		p, _ := strconv.Atoi(strings.TrimSpace(port))
+		name = strings.TrimSpace(strings.ReplaceAll(name, "\"", ""))
+		protocolName = strings.TrimSpace(strings.ReplaceAll(protocolName, "\"", ""))
+		noderes := PortDeclared{port: p, name: name, protocol: protocolName}
+		result = append(result, noderes)
+		port, name, protocolName = "", "", ""
+	}
+	return result
+}
+
+func getListeningPortList(temp []string) []PortListen {
+	listeningPortCommand := strings.Join(temp, " ")
+	fmt.Println(listeningPortCommand)
+	res, _ := utils.ExecuteCommand(listeningPortCommand, ocCommandTimeOut, interactive.GetContext(false))
+	lines := strings.Split(res, "\n")
+	var result []PortListen
+	for _, line := range lines {
+		fields := strings.Fields(line)
+		if indexProtocolName > len(fields) || indexPort > len(fields) {
+			return result
+		}
+		s := strings.Split(fields[indexPort], ":")
+		p, _ := strconv.Atoi(s[1])
+		nodeListen := PortListen{port: p, protocol: fields[indexProtocolName]}
+		result = append(result, nodeListen)
+	}
+	return result
+}
+
+func checkIfListenIsDeclared(declaredPortList []PortDeclared, listeningPortList []PortListen) bool {
+	indexElement1 := 0
+	indexElement2 := 0
+
+	for indexElement1 < len(declaredPortList) && indexElement2 < len(listeningPortList) {
+		if declaredPortList[indexElement1].port == listeningPortList[indexElement2].port {
+			indexElement1++
+			indexElement2++
+			continue
+		}
+		if listeningPortList[indexElement2].port < declaredPortList[indexElement1].port {
+			return false
+		}
+		indexElement1++
+	}
+	return true
+}
+
+func testListenAndDeclared(env *config.TestEnvironment) {
+	var declaredPort []PortDeclared
+	var listeningPort []PortListen
+	var x *config.Container
+	res := true
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestServicesDoNotUseNodeportsIdentifier)
+	ginkgo.It(testID, func() {
+		for _, podUnderTest := range env.PodsUnderTest {
+			for i := 0; i < podUnderTest.ContainerCount; i++ {
+				temp := getDeclaredPortList(i, podUnderTest.Name, podUnderTest.Namespace)
+				declaredPort = append(declaredPort, temp...)
+			}
+
+			nodeName := podnodename.NewPodNodeName(common.DefaultTimeout, podUnderTest.Name, podUnderTest.Namespace)
```

Comment:
the nodename information is store inside container, using a handler to get this information is an overkill, you can add another loop to get it.